### PR TITLE
reference/best-practices: improve the wording and format of HAProxy Best Practices doc

### PR DESCRIPTION
**Pull request by juliezhang1112**
### What is changed, added or deleted? 
- Improve the wording and format of HAProxy Best Practices doc
- Correct command-line options and their descriptions
- Move the placement of an image

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1889

### Which version does your change affect? 
dev, v3.0, v3.1, v2.1

### CODE REVIEW

**Comment by juliezhang1112** [comment]
PTAL @lilin90 Thanks~

File: dev/reference/best-practices/haproxy.md [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
-[HAProxy](https://github.com/haproxy/haproxy) 是由 C 语言编写的自由开放源码的软件，为基于 TCP 和 HTTP 协议的应用程序提供高可用性、负载均衡和代理服务。因为 HAProxy 能够快速、高效使用 CPU 和内存，所以目前使用非常广泛，许多知名网站诸如 GitHub、Bitbucket、Stack Overflow、Reddit、Tumblr、Twitter 和 Tuenti 以及亚马逊网络服务系统都在使用 HAProxy。
+HAProxy 是由 C 语言编写的自由开放源码的软件，为基于 TCP 和 HTTP 协议的应用程序提供高可用性、负载均衡和代理服务。因为 HAProxy 能够快速、高效使用 CPU 和内存，所以目前使用非常广泛，许多知名网站诸如 GitHub、Bitbucket、Stack Overflow、Reddit、Tumblr、Twitter 和 Tuenti 以及亚马逊网络服务系统都在使用 HAProxy。
 
 HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，他现在仍然负责该项目的维护，并在开源社区免费提供版本迭代。最新的稳定版本 2.0.0 于 2019 年 8 月 16 日发布，带来更多[优秀的特性](https://www.haproxy.com/blog/haproxy-2-0-and-beyond/)。
 
 ## HAProxy 部分核心功能介绍
 
 - [高可用性](http://cbonte.github.io/haproxy-dconv/1.9/intro.html#3.3.4)：HAProxy 提供优雅关闭服务和无缝切换的高可用功能；
-- [负载均衡](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html#4.2-balance)：L4（TCP）和 L7（HTTP）负载均衡模式，至少 9 类均衡算法，比如 roundrobin，leastconn，random 等；
+- [负载均衡](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html#4.2-balance)：L4（TCP）和 L7（HTTP）两种负载均衡模式，至少 9 类均衡算法，比如 roundrobin，leastconn，random 等；
```
Comment (lilin90) [inline dev/reference/best-practices/haproxy.md]:
```suggestion
- [负载均衡](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html#4.2-balance)：L4 (TCP) 和 L7 (HTTP) 两种负载均衡模式，至少 9 类均衡算法，比如 roundrobin，leastconn，random 等；
```

File: dev/reference/best-practices/haproxy.md [hunk 2]
```diff
@@ -67,143 +73,152 @@ yum -y install epel-release gcc systemd-devel
...
     which haproxy
     ```
 
+    ```
+    /usr/sbin/haproxy
+    ```
+
 #### HAProxy 命令介绍
 
+执行如下命令查看命令行参数及基本用法:
```
Comment (lilin90) [inline dev/reference/best-practices/haproxy.md]:
```suggestion
执行如下命令查看命令行参数及基本用法：
```

**Comment by juliezhang1112** [comment]
Updated. PTAL again~ Thanks! @lilin90

File: dev/reference/best-practices/haproxy.md [hunk 3]
```diff
@@ -67,143 +73,152 @@ yum -y install epel-release gcc systemd-devel
...
+| `-C <dir>` | 在加载配置文件之前更改目录位置至 \<dir>。 |
+| `-W` | 主从模式。 |
+| `-q` | 静默模式，不输出信息。 |
+| `-c` | 只检查配置文件并在尝试绑定之前退出。 |
+| `-n <limit>` | 设置每个进程的最大总连接数为 \<limit>。 |
+| `-m <limit>` | 设置所有进程的最大可用内存为 \<limit>（单位：MB）。 |
+| `-N <limit>` | 设置单点最大连接数为 \<limit>，默认为 2000。 |
+| `-L <name>` | 将本地实例对等名称改为 \<name>，默认为本地主机名。 |
+| `-p <file>` | 将 HAProxy 所有子进程的 PID 信息写入 \<file>。 |
+| `-de` | 禁止使用 epoll(7)，epoll(7)仅在 Linux 2.6 和某些定制的 Linux 2.4 系统上可用。 |
```
Comment (lilin90) [inline dev/reference/best-practices/haproxy.md]:
```suggestion
| `-de` | 禁止使用 epoll(7)，epoll(7) 仅在 Linux 2.6 和某些定制的 Linux 2.4 系统上可用。 |
```

File: dev/reference/best-practices/haproxy.md [hunk 4]
```diff
@@ -67,143 +73,152 @@ yum -y install epel-release gcc systemd-devel
...
+| `-dp` | 禁止使用 epoll(2)，可改用 select(2)。 |
+| `-dS` | 禁止使用 splice(2)，splice(2)在一些旧版 Linux 内核上不可用。 |
```
Comment (lilin90) [inline dev/reference/best-practices/haproxy.md]:
```suggestion
| `-dS` | 禁止使用 splice(2)，splice(2) 在一些旧版 Linux 内核上不可用。 |
```

File: dev/reference/best-practices/haproxy.md [hunk 5]
```diff
@@ -67,143 +73,152 @@ yum -y install epel-release gcc systemd-devel
...
+| `-dR` | 禁止使用 SO_REUSEPORT。 |
+| `-dr` | 忽略服务器地址解析失败。 |
+| `-dV` | 禁止在服务器端使用 SSL。 |
+| `-sf <pidlist>` | 启动后，向 pidlist 中的 PID 发送"finish"信号，收到此信号的进程在退出之前将等待所有会话完成，即优雅停止服务。此选项必须最后指定，后跟任意数量的 PID。从技术上讲，SIGTTOU 和 SIGUSR1 都被发送。 |
```
Comment (lilin90) [inline dev/reference/best-practices/haproxy.md]:
```suggestion
| `-sf <pidlist>` | 启动后，向 pidlist 中的 PID 发送 "finish" 信号，收到此信号的进程在退出之前将等待所有会话完成，即优雅停止服务。此选项必须最后指定，后跟任意数量的 PID。从技术上讲，SIGTTOU 和 SIGUSR1 都被发送。 |
```

File: dev/reference/best-practices/haproxy.md [hunk 6]
```diff
@@ -67,143 +73,152 @@ yum -y install epel-release gcc systemd-devel
...
+| `-st <pidlist>` | 启动后，向 pidlist 中的 PID 发送"terminate"信号，收到此信号的进程将立即终止，关闭所有活动会话。此选项必须最后指定，后跟任意数量的 PID。从技术上讲，SIGTTOU 和 SIGTERM 都被发送。 |
```
Comment (lilin90) [inline dev/reference/best-practices/haproxy.md]:
```suggestion
| `-st <pidlist>` | 启动后，向 pidlist 中的 PID 发送 "terminate" 信号，收到此信号的进程将立即终止，关闭所有活动会话。此选项必须最后指定，后跟任意数量的 PID。从技术上讲，SIGTTOU 和 SIGTERM 都被发送。 |
```

**Pull request merged**